equinix/resource_fabric_service_profile.go

```diff
@@ -1114,13 +1109,13 @@ func marketingInfoToFabric(schemaMarketingInfo []interface{}) v4.MarketingInfo {
 		return v4.MarketingInfo{}
```

Comment:
The method should return a `*v4.MarketingInfo`, and if the input is `nil` then we should return `nil`.